addons/account/models/account_payment.py

```diff
@@ -249,72 +249,41 @@ def _get_valid_liquidity_accounts(self):
             self.journal_id.outbound_payment_method_line_ids.payment_account_id
         )
 
-    def _get_aml_default_display_map(self):
-        return {
-            ('outbound', 'customer'): _("Customer Reimbursement"),
-            ('inbound', 'customer'): _("Customer Payment"),
-            ('outbound', 'supplier'): _("Vendor Payment"),
-            ('inbound', 'supplier'): _("Vendor Reimbursement"),
-        }
-
     def _get_aml_default_display_name_list(self):
         """ Hook allowing custom values when constructing the default label to set on the journal items.
 
         :return: A list of terms to concatenate all together. E.g.
             [
-                ('label', "Vendor Reimbursement"),
-                ('sep', ' '),
-                ('amount', "$ 1,555.00"),
-                ('sep', ' - '),
-                ('date', "05/14/2020"),
+                ('payment_method', "Greg's Card"),
+                ('sep', ": "),
+                ('memo', "New Computer"),
             ]
         """
         self.ensure_one()
-        display_map = self._get_aml_default_display_map()
-        values = [
-            ('label', _("Internal Transfer") if self.is_internal_transfer else display_map[(self.payment_type, self.partner_type)]),
```

Comment:
Beware that nothing is said for the internal transfers in the specs. You need to check with a PO what to do in that case